### DISCUSSION TITLE
Possibility to add new webauthn device types.

### DISCUSSION THREAD

**Discussion by lohkee27** [goauthentik/authentik#10104] [Ideas]
While trying to add latest Yubico Yubikeys with firmware 5.7 to the list of allowed device types (https://docs.goauthentik.io/docs/flow/stages/authenticator_webauthn/#device-type-restrictions), I cannot find them in the list of available device types.

It would be a nice option for an admin to add a new device type to the list of available device types, so it can be added to the selected device types list.

I can also see that there is an option to add unknown devices to the selected list, but it is not clear how does that change the behavior then.

Would that mean that the selected device types and any unknown device, that is not in the available device types list, are allowed?

**Comment by BeryJu** [comment]
I don't think a specific version of firmware on yubikeys would get a new AAGUID? But regardless, yeah we should add the option to add custom device types

**Reply by lohkee27** [reply]
Thank you for a quick reply. But according to this page they get new AAGUIDs:
https://support.yubico.com/hc/en-us/articles/360016648959-YubiKey-Hardware-FIDO2-AAGUIDs
